### PR TITLE
Refactor deployment templates to support additional env vars from Secret

### PR DESCRIPTION
**Pull request by debugger24**
## What was changed
Added option to pass secret name which will be mouted as env variables in the contianer. 

## Why?
This will help in scenarios like SSO integration or other conifgurations.

## Checklist

1. Closes 
https://github.com/temporalio/helm-charts/issues/236

2. How was this tested:
Using helm render locally.

3. Any docs updates needed?
Added `Enable SSO in Temporal UI` section in README.md

### CODE REVIEW

File: charts/temporal/values.yaml [hunk 1]
```diff
@@ -263,6 +263,7 @@ server:
     tolerations: []
     affinity: {}
     additionalEnv: []
+    additionalEnvSecretName: ""
```
Comment (robholland) [inline charts/temporal/values.yaml:266]: What do you think about `additionalEnvFromSecret`?

Comment (robholland) [reply charts/temporal/values.yaml:266]: The value would be the name.

Comment (debugger24) [reply charts/temporal/values.yaml:266]: `additionalEnvSecretName` is more clear it needs secret name here.

File: charts/temporal/templates/server-deployment.yaml [hunk 2]
```diff
@@ -88,6 +88,11 @@ spec:
           {{- if or $.Values.server.additionalEnv $serviceValues.additionalEnv }}
           {{- toYaml (default $.Values.server.additionalEnv $serviceValues.additionalEnv) | nindent 12 }}
           {{- end }}
+          {{- if.Values.server.additionalEnvSecretName }}
```
Comment (robholland) [inline charts/temporal/templates/server-deployment.yaml:91]:
```suggestion
          {{- if .Values.server.additionalEnvSecretName }}
```

File: charts/temporal/templates/admintools-deployment.yaml [hunk 3]
```diff
@@ -39,6 +39,11 @@ spec:
           {{- if .Values.admintools.additionalEnv }}
           {{- toYaml .Values.admintools.additionalEnv | nindent 12 }}
           {{- end }}
+          {{- if.Values.admintools.additionalEnvSecretName }}
```
Comment (robholland) [inline charts/temporal/templates/admintools-deployment.yaml:42]:
```suggestion
          {{- if .Values.admintools.additionalEnvSecretName }}
```

File: charts/temporal/templates/web-deployment.yaml [hunk 4]
```diff
@@ -36,6 +36,11 @@ spec:
           {{- if .Values.web.additionalEnv }}
           {{- toYaml .Values.web.additionalEnv | nindent 12 }}
           {{- end }}
+          {{- if.Values.web.additionalEnvSecretName }}
```
Comment (robholland) [inline charts/temporal/templates/web-deployment.yaml:39]:
```suggestion
          {{- if .Values.web.additionalEnvSecretName }}
```

**Review by robholland — CHANGES_REQUESTED** [review]

**Review by robholland — APPROVED** [review]

**Pull request merged**